src/main/java/com/fasterxml/jackson/databind/ser/std/MapSerializer.java

```diff
@@ -1158,6 +1158,28 @@ protected final JsonSerializer<Object> _findAndAddDynamic(PropertySerializerMap
         return result.serializer;
     }
 
+    // Since 2.19
+    protected Map<?, ?> _tryOrderingEntries(Map<?,?> input, JsonGenerator gen,
+            SerializerProvider provider)
+        throws IOException
+    {
+        try {
+            return _orderEntries(input, gen, provider);
+        } catch (ClassCastException cce) {
+            // [databind#4773] Since 2.19, `SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS` should not
+            // apply to Maps with incomparable keys.
+            // Due to dynamic nature of Map, we cannot be able to check key type in advance, so
+            // we may catch ClassCastException and handle it gracefully.
+            if (cce.getMessage() != null
+                && cce.getMessage().contains("cannot be cast to class java.lang.Comparable")
```

Comment:
Hmmh. I don't like catching exception in general, but specifically I dislike relying on exception message here. :-/